.azure-pipelines.yml

```diff
@@ -0,0 +1,89 @@
+#
+# Azure Pipelines configuration for building and testing react on Linux, Windows, and macOS.
+#
+
+trigger:
+- master
+
+jobs:
+- job: LinuxBasic
+  displayName: 'Linux: Basic Tests'
+  pool:
+    vmImage: 'ubuntu-16.04'
+  steps:
+    - template: .azure-pipelines/common/setup.yml
+    - template: .azure-pipelines/linux/basic-tests.yml
+    - template: .azure-pipelines/common/publish-error-codes.yml
+      parameters:
+        artifactName: 'error-codes-linux-basic'
+
+- job: LinuxProd
+  displayName: 'Linux: Prod Tests'
+  pool:
+    vmImage: 'ubuntu-16.04'
+  steps:
+    - template: .azure-pipelines/common/setup.yml
+    - template: .azure-pipelines/linux/prod-tests.yml
+    - template: .azure-pipelines/common/publish-error-codes.yml
+      parameters:
+        artifactName: 'error-codes-linux-prod'
+
+- job: LinuxFire
+  displayName: 'Linux: React Fire Tests'
+  pool:
+    vmImage: 'ubuntu-16.04'
+  steps:
+    - template: .azure-pipelines/common/setup.yml
+    - template: .azure-pipelines/linux/fire-tests.yml
+    - template: .azure-pipelines/common/publish-error-codes.yml
+      parameters:
+        artifactName: 'error-codes-linux-fire'    
+
+- job: LinuxArtifacts
+  displayName: 'Linux: Build and Upload Artifacts'
+  pool:
+    vmImage: 'ubuntu-16.04'
+  steps:
+    - template: .azure-pipelines/common/setup.yml
+    - template: .azure-pipelines/linux/build-and-upload-artifacts.yml
+    - template: .azure-pipelines/common/publish-error-codes.yml
+      parameters:
+        artifactName: 'error-codes-linux-artifacts'        
+
+- job: LinuxCoverage
+  displayName: 'Linux: Test Coverage'
+  pool:
+    vmImage: 'ubuntu-16.04'
+  steps:
+    - template: .azure-pipelines/common/setup.yml
+    - template: .azure-pipelines/linux/test-coverage.yml
+
+- job: WindowsTests
+  displayName: 'Windows: Basic Tests'
+  pool:
+    vmImage: 'vs2017-win2016'
+  steps:
+    - template: .azure-pipelines/common/setup.yml
+    - script: |
+        git config core.autocrlf input
+        git reset HEAD --hard
+      displayName: 'Fix line endings in Windows'
+    - template: .azure-pipelines/windows/basic-tests.yml
+    - template: .azure-pipelines/common/publish-error-codes.yml
+      parameters:
+        artifactName: 'error-codes-windows-basic'            
+
+- job: Lint
+  displayName: 'Lint'
+  pool:
+    vmImage: 'ubuntu-16.04'
+  steps:
+    - template: .azure-pipelines/common/setup.yml
+    - script: yarn lint
+      displayName: 'Run lint'
+
+  variables:
+    AZURE_PIPELINES_BRANCH: $(Build.SourceBranchName)
```

Comment:
I **heard** the value of the "source branch name" environment variable is only the last segment of a branch that has `/` separators. So, for example, for branch `users/wismythe/foo`, the value of the variable will be `foo. This would be incredibly lame if it was the case, but you should double-check, and ensure their build logic won't get screwed up because of this ....

Reply:
You are correct. So for PR's, it always shakes out to `merge`. They only use the branch name to do a check on if it's `master` so it should be ok.